### DISCUSSION TITLE
How do I do a "shallow" or non-recursive dump of a model instance in v2?

### DISCUSSION THREAD

**Discussion by spyoungtech** [pydantic/pydantic#8257] [Question]
In v1, I was using the following code to generate "shallow" dumps of a model:

```py
import pydantic # v1.10
class MyBaseModel(pydantic.BaseModel):
    def shallow_dict(self, by_alias: bool = True, exclude_unset: bool = True, exclude_defaults: bool = False):
        """
        Like ``.dict`` but not recursive
        """
        d = dict(self._iter(
            to_dict=False,
            by_alias=by_alias,
            exclude_unset=exclude_unset,
            include=None,
            exclude=None,
            exclude_defaults=exclude_defaults,
            exclude_none=False
        ))
        return d
```

For example:

```py
class InnerModel(MyBaseModel):
    foo: int = pydantic.Field(1)

class OuterModel(MyBaseModel):
    inner: InnerModel = pydantic.Field(...)
    bar: str = pydantic.Field(...)


model = OuterModel(inner=InnerModel(), bar='bacon')
print(model.shallow_dict())
# {'inner': InnerModel(foo=1), 'bar': 'bacon'}
```

So, to migrate my project to v2, I'd like to make a `shallow_dump` method.

In v2 I'm glad to see the options for `._iter` (the underlying method for `.dict`) are exposed publicly in `model_dump`. However,  it's not clear to me how I can prevent `model_dump` in v2 from being recursive.

```py
import pydantic # v2.5
class MyBaseModel(pydantic.BaseModel):
    def shallow_dump(self, by_alias: bool = True, exclude_unset: bool = True, exclude_defaults: bool = False):
        """
        Like ``.model_dump`` but not recursive
        """
       # ... ?
```

Technically using `._iter` is still working, but is considered deprecated:

>  PydanticDeprecatedSince20: The private method `_iter` will be removed and should no longer be used. Deprecated in Pydantic V2.0 to be removed in V3.0.

So I'm wondering how I can replace the use of this method for my use case. Thanks!

**Comment by Lemminkyinen** [comment]
Maybe something like this? This does support `@computed_field` as well

```python
def shallow_dump(self) -> Dict[str, Any]:
    keys = self.model_dump().keys()
    get_attr = partial(getattr, self)
    return {key: get_attr(key) for key in keys}
```

**Comment by MikuAuahDark** [comment]
I was looking on how to do this in Google if there's something built-in. Apparently not. So I create my own.

This solution supports `@computed_field` and does not call `model_dump()` which should improve the performance.

```py
from itertools import chain

def shallow_dump(model: pydantic.BaseModel, /):
    keys = chain(model.__class__.model_fields.keys(), model.model_computed_fields.keys())
    return {k: getattr(model, k) for k in keys}
```

**Comment by cc-stjm** [comment]
Unfortunately the variants above don't obey aliases. If you want to obey aliases, here's my attempt:

```
    def shallow_dump(self, /):
        fields = chain(self.__class__.model_fields.items(), self.model_computed_fields.items())
        return {field[1].serialization_alias or field[1].alias or field[0]: getattr(self, field[0]) for field in fields}
```
Although this still doesn't do everything `model_dump` does (e.g. some fields shouldn't be serialized)
What I'd really love is a version of `model_dump` that had a `depth` parameter, and would leave objects as pydantic models once it'd iterated to that depth. (So essentially this'd be depth=1)
